### DISCUSSION TITLE
rectangle has no sandbox entitlements

### DISCUSSION THREAD

**Discussion by faced0a** [rxhanson/Rectangle#656] [General]
Hello - 

Is there some strong reason that rectangle does not include sandbox entitlements ?

    <key>com.apple.security.app-sandbox</key>
    <true/>

I just noticed it's running without sandbox.

**Comment by rxhanson** [comment]
Apps that manage windows cannot be sandboxed because they manage windows of other applications, via the Accessibility API. A long time ago, I thought there might be a chance to sandbox it to get it on the App Store, since Magnet, Moom, and BetterSnapTool were on there, but those apps were grandfathered in from before Apple restricted it.

**Comment by rxhanson** [comment]
1 Yes, Apple is aware of this limitation, as they made it this way. 

2 The security risk at stake here is that at most, Rectangle could use accessibility permissions to log keystrokes somewhere. Obviously, Rectangle does not do that. The only way you could end up in a bad scenario with Rectangle is if you installed and used a version of it that was tampered with by a bad actor, to do such a thing. So long as you download from this repository / https://rectangleapp.com, then you won’t have this issue. 

In my opinion, Apple should separate it’s window management API from the accessibility API, and create an entitlement that can be used to allow said window management API in the sandbox environment. By all means, feel free to open an issue Apple side for this. I haven’t done it.

**Comment by faced0a** [comment]
isn't com.apple.security.automation.apple-events enough?

**Reply by rxhanson** [reply]
Unfortunately not, since the window management api available to 3rd party developers is part of the accessibility api.
